diff --git a/node_modules/@chainlink/contracts/src/v0.6/VRFConsumerBase.sol b/node_modules/@chainlink/contracts/src/v0.6/VRFConsumerBase.sol
index e90adbb..c89a67b 100644
--- a/node_modules/@chainlink/contracts/src/v0.6/VRFConsumerBase.sol
+++ b/node_modules/@chainlink/contracts/src/v0.6/VRFConsumerBase.sol
@@ -1,5 +1,5 @@
 // SPDX-License-Identifier: MIT
-pragma solidity ^0.6.0;
+pragma solidity >0.6.0;
 
 import "./vendor/SafeMathChainlink.sol";
 
@@ -179,7 +179,7 @@ abstract contract VRFConsumerBase is VRFRequestIDBase {
    *
    * @dev https://docs.chain.link/docs/link-token-contracts
    */
-  constructor(address _vrfCoordinator, address _link) public {
+  constructor(address _vrfCoordinator, address _link) {
     vrfCoordinator = _vrfCoordinator;
     LINK = LinkTokenInterface(_link);
   }
diff --git a/node_modules/@chainlink/contracts/src/v0.6/VRFRequestIDBase.sol b/node_modules/@chainlink/contracts/src/v0.6/VRFRequestIDBase.sol
index 2668ead..eb0a6e5 100644
--- a/node_modules/@chainlink/contracts/src/v0.6/VRFRequestIDBase.sol
+++ b/node_modules/@chainlink/contracts/src/v0.6/VRFRequestIDBase.sol
@@ -1,5 +1,5 @@
 // SPDX-License-Identifier: MIT
-pragma solidity ^0.6.0;
+pragma solidity >0.6.0;
 
 contract VRFRequestIDBase {
 
diff --git a/node_modules/@chainlink/contracts/src/v0.6/interfaces/LinkTokenInterface.sol b/node_modules/@chainlink/contracts/src/v0.6/interfaces/LinkTokenInterface.sol
index eeb9444..7803276 100644
--- a/node_modules/@chainlink/contracts/src/v0.6/interfaces/LinkTokenInterface.sol
+++ b/node_modules/@chainlink/contracts/src/v0.6/interfaces/LinkTokenInterface.sol
@@ -1,5 +1,5 @@
 // SPDX-License-Identifier: MIT
-pragma solidity ^0.6.0;
+pragma solidity >0.6.0;
 
 interface LinkTokenInterface {
   function allowance(address owner, address spender) external view returns (uint256 remaining);
diff --git a/node_modules/@chainlink/contracts/src/v0.6/vendor/SafeMathChainlink.sol b/node_modules/@chainlink/contracts/src/v0.6/vendor/SafeMathChainlink.sol
index 39d73a5..77b5625 100644
--- a/node_modules/@chainlink/contracts/src/v0.6/vendor/SafeMathChainlink.sol
+++ b/node_modules/@chainlink/contracts/src/v0.6/vendor/SafeMathChainlink.sol
@@ -1,5 +1,5 @@
 // SPDX-License-Identifier: MIT
-pragma solidity ^0.6.0;
+pragma solidity >0.6.0;
 
 /**
  * @dev Wrappers over Solidity's arithmetic operations with added overflow
