// @chainlink/contracts/ethers/v0.7/factories/UpkeepCompatible__factory.ts
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */

import { Contract, Signer } from "ethers";
import { Provider } from "ethers/providers";

import { UpkeepCompatible } from "../UpkeepCompatible";

export class UpkeepCompatible__factory {
  static connect(
    address: string,
    signerOrProvider: Signer | Provider
  ): UpkeepCompatible {
    return new Contract(address, _abi, signerOrProvider) as UpkeepCompatible;
  }
}

const _abi = [
  {
    inputs: [
      {
        internalType: "bytes",
        name: "data",
        type: "bytes",
      },
    ],
    name: "checkForUpkeep",
    outputs: [
      {
        internalType: "bool",
        name: "",
        type: "bool",
      },
      {
        internalType: "bytes",
        name: "",
        type: "bytes",
      },
    ],
    stateMutability: "nonpayable",
    type: "function",
  },
  {
    inputs: [
      {
        internalType: "bytes",
        name: "data",
        type: "bytes",
      },
    ],
    name: "performUpkeep",
    outputs: [],
    stateMutability: "nonpayable",
    type: "function",
  },
];
