// @chainlink/contracts/ethers/v0.6/factories/Reverter__factory.js
"use strict";
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */
Object.defineProperty(exports, "__esModule", { value: true });
const ethers_1 = require("ethers");
class Reverter__factory extends ethers_1.ContractFactory {
    constructor(signer) {
        super(_abi, _bytecode, signer);
    }
    deploy(overrides) {
        return super.deploy(overrides);
    }
    getDeployTransaction(overrides) {
        return super.getDeployTransaction(overrides);
    }
    attach(address) {
        return super.attach(address);
    }
    connect(signer) {
        return super.connect(signer);
    }
    static connect(address, signerOrProvider) {
        return new ethers_1.Contract(address, _abi, signerOrProvider);
    }
}
exports.Reverter__factory = Reverter__factory;
const _abi = [
    {
        stateMutability: "payable",
        type: "fallback",
    },
];
const _bytecode = "0x6080604052348015600f57600080fd5b50609280601d6000396000f3fe608060408190527f08c379a00000000000000000000000000000000000000000000000000000000081526020608452601660a4527f5261697365642062792052657665727465722e736f6c0000000000000000000060c452606490fdfea2646970667358221220792e3ee1ea1d3b8da46487a15a190d9756bcb4c30a130a99c7275483873b12f364736f6c63430006060033";
